app/models/manageiq/providers/nuage/network_manager/event_catcher/stream.rb

```diff
@@ -0,0 +1,46 @@
+require 'qpid_proton'
+
+class ManageIQ::Providers::Nuage::NetworkManager::EventCatcher::Stream
+  def self.test_amqp_connection(options = {})
+    # Ensure we just test the connection. AMQP channel will be established and
+    # started, however it will be immediately stopped.
+    options[:test_connection] = true
```

Comment:
Yes, unfortunately it works like this. In case the URL is incorrect, this will be noticed immediately when opening the connection, but in case just the `user:pass` is incorrect, we need to wait for the connection to be opened - otherwise the corresponding error event will be thrown.